src/cargo/util/command_prelude.rs

```diff
@@ -986,6 +995,40 @@ pub fn root_manifest(manifest_path: Option<&Path>, gctx: &GlobalContext) -> Carg
     }
 }
 
+pub fn get_examples_from_cwd() -> CargoResult<Vec<OsString>> {
+    // Get the current directory
+    let cwd = std::env::current_dir()?;
+    let mut examples = vec![];
+
+    // Get all files in the `examples` directory with a `.rs` extension
+    let examples_dir = cwd.join("examples");
+    if examples_dir.is_dir() {
```

Comment:
Again, we should be using Cargo's APIs to look up the actual example names, rather than trying to guess